src/main/java/org/wwarn/vivax/manager/web/rest/dto/SiteDataViewDTO.java

```diff
@@ -103,88 +104,103 @@ public void setListPublications(Set<Publication> listPublications) {
-				+ location + ", listTreatments=" + listTreatments + ", listPublications=" + listPublications + "]";
-	}
-
-	@Override
-	public int hashCode() {
-		final int prime = 31;
-		int result = 1;
-		result = prime * result + ((category == null) ? 0 : category.hashCode());
-		result = prime * result + (int) (id ^ (id >>> 32));
-		result = prime * result + ((listPublications == null) ? 0 : listPublications.hashCode());
-		result = prime * result + ((listTreatments == null) ? 0 : listTreatments.hashCode());
-		result = prime * result + ((location == null) ? 0 : location.hashCode());
-		result = prime * result + ((ref == null) ? 0 : ref.hashCode());
-		result = prime * result + ((typeStudy == null) ? 0 : typeStudy.hashCode());
-		result = prime * result + ((upper95CI == null) ? 0 : upper95CI.hashCode());
-		result = prime * result + ((yearEnd == null) ? 0 : yearEnd.hashCode());
-		result = prime * result + ((yearStart == null) ? 0 : yearStart.hashCode());
-		return result;
-	}
-
-	@Override
-	public boolean equals(Object obj) {
-		if (this == obj)
-			return true;
-		if (obj == null)
-			return false;
-		if (getClass() != obj.getClass())
-			return false;
-		SiteDataViewDTO other = (SiteDataViewDTO) obj;
-		if (category == null) {
-			if (other.category != null)
-				return false;
-		} else if (!category.equals(other.category))
-			return false;
-		if (id != other.id)
-			return false;
-		if (listPublications == null) {
-			if (other.listPublications != null)
-				return false;
-		} else if (!listPublications.equals(other.listPublications))
-			return false;
-		if (listTreatments == null) {
-			if (other.listTreatments != null)
-				return false;
-		} else if (!listTreatments.equals(other.listTreatments))
-			return false;
-		if (location == null) {
-			if (other.location != null)
-				return false;
-		} else if (!location.equals(other.location))
-			return false;
-		if (ref == null) {
-			if (other.ref != null)
-				return false;
-		} else if (!ref.equals(other.ref))
-			return false;
-		if (typeStudy == null) {
-			if (other.typeStudy != null)
-				return false;
-		} else if (!typeStudy.equals(other.typeStudy))
-			return false;
-		if (upper95CI == null) {
-			if (other.upper95CI != null)
-				return false;
-		} else if (!upper95CI.equals(other.upper95CI))
-			return false;
-		if (yearEnd == null) {
-			if (other.yearEnd != null)
-				return false;
-		} else if (!yearEnd.equals(other.yearEnd))
-			return false;
-		if (yearStart == null) {
-			if (other.yearStart != null)
-				return false;
-		} else if (!yearStart.equals(other.yearStart))
-			return false;
-		return true;
-	}
+    public SiteDataViewDTO(long id, String typeStudy, String upper95CI, Integer yearStart, Integer yearEnd, String category, String ref, Location location, Set<Treatment> listTreatments, Set<Publication> listPublications, List <String> listTreatmentArmCodes, List <Integer> listPubMedIds) {
+        this.id = id;
```

Comment:
this is a massive contructor, if you are gonna need more than 3 params in your class, think about using the java builder pattern: https://jlordiales.wordpress.com/2012/12/13/the-builder-pattern-in-practice/